[IMP] surveys: time_random rewrite for questions (#8259)
odoo/documentation: content/applications/marketing/surveys/time_random.rst

`Enter` or click away.

The section line appears in dark gray in the :guilabel:`Questions` tab.

Then, drag-and-drop desired questions beneath a section, or drag-and-drop a section title on top of
(i.e. *before*) the desired question(s) in the survey, in order to populate the section with
specific questions that fit the theme of the section.

If the :guilabel:`Randomized per Section` option is enabled in the :guilabel:`Options` tab of the
survey form, the number `1` appears, by default, on the section line, beneath the :guilabel:`#
Questions Randomly Picked` column.

This indicates that `1` question from that section will be picked at random for each participant
taking the survey, bypassing every other, non-chosen question from that section. To change that
number, select that figure, and type in the desired number in its place. Then, either press `Enter`,
or click away.

Create questions
================

To create questions for the survey, click :guilabel:`Add a question` in the :guilabel:`Questions`
tab.

Clicking :guilabel:`Add a question` opens the :guilabel:`Create Sections and Questions` pop-up

> window, in which a survey question can be created.

Author: considering this suggestion (re: "modal") was made prior to our collective agreement to go with "pop-ups" over "modal", I am going to ignore this (and all the other similar comments on the PR)

================
Create questions
================

> When it comes to :doc:`creating and customizing surveys <../surveys/create>` in the Odoo *Surveys*
> application, the creation and customization of questions for that survey are of the utmost
> importance.

Reviewer: ```suggestion
In Odoo *Surveys*, crafting and tailoring survey questions is crucial for :doc:`creating and customizing surveys <../surveys/create>`.
```

Fortunately, Odoo provides numerous ways to configure tailored questions for any kind of survey.

To access a list of *all* the questions that have been created in the database, navigate to
:menuselection:`Surveys app --> Questions & Answers --> Questions`. There, users can view and modify
any question from any survey.

However, there is only one place in the Odoo *Surveys* application where survey questions can be
created. To do that, navigate to a survey form, by going to :menuselection:`Surveys app --> New`, or
by selecting any pre-existing survey from the :guilabel:`Surveys` page (:menuselection:`Surveys app

> --> Surveys`).

Reviewer: As this is a lengthy doc (and I'm sure users will be super grateful that you're going into so much detail for each question type), but I'm wondering if there's a way to summarize the info in a chart with links. Something like my list of removal strategies and their features [here](https://www.odoo.com/documentation/17.0/applications/inventory_and_mrp/inventory/warehouses_storage/removal_strategies.html)

Since you provided a lot of details for people to jump-to sections in the header, and users can jump to sections using the side navigation bar. So, I think this suggestion is a small minor thing and can be adjusted in a follow-up PR for further improvement if you wish (:

For the purposes of THIS PR, feel free to ignore. Just wanted to make a comment (:

email` appears. If that box is ticked, Odoo saves the participant's answer to that specific question
as their email address.

If the :guilabel:`Save as user nickname` option is enabled, Odoo saves the participant's answer as
its nickname.

For all other applicable :guilabel:`Question Type` options that provide answer options to the
participant, the :guilabel:`Answers` tab appears the same.

.. image:: time_random/multiple-choice-answer-options.png
   :align: center
   :alt: The answer tab with a multiple choice answer option selected in Odoo Surveys.

From here, users can add answer options by clicking :guilabel:`Add a line`, and typing in the
various answer options for that question. Then, either press `Enter` to lock in that answer option
and immmediately add another one. Or, click away to simply lock in that answer option.

The entered answer options appear in the :guilabel:`Choices` column of the :guilabel:`Answers` tab.

If any *Scoring* option is enabled in the :guilabel:`Options` tab of the survey form, the
:guilabel:`Correct` and :guilabel:`Score` columns appear to the right of the :guilabel:`Choices`
column.

> If an answer option is the correct option, designate it as such by ticking the box beneath the
> :guilabel:`Correct` column on the corresponding question line. If :guilabel:`Multiple choice:
> multiple answers allowed` is set as the :guilabel:`Question Type` more than one answer option in the
> :guilabel:`Choices` column can be marked as :guilabel:`Correct`.

Reviewer: ```suggestion
To mark an answer option as correct, tick the box beneath the
:guilabel:`Correct` column for the respective question. If :guilabel:`Multiple choice:
multiple answers allowed` is set as the :guilabel:`Question Type`, multiple answers in the
:guilabel:`Choices` column can be marked as :guilabel:`Correct`.
```

section of the :guilabel:`Options` tab: :guilabel:`Validate Entry` and :guilabel:`Placeholder`.

If the :guilabel:`Validate Entry` option is enabled, two additional fields appear beneath:
:guilabel:`Min/Max Limits` and :guilabel:`Validation Error`.

.. image:: time_random/single-line-validate-entry.png
   :align: center
   :alt: The answers section of the option tab when a single line text box question type is chosen.

In the :guilabel:`Min/Max Limits` field, designate the minimum and maxium allowed quantities for
that specific question.

In the :guilabel:`Validation Error` field, enter a custom message that Odoo displays when an answer
is not valid.

In the :guilabel:`Placeholder` field, enter a guiding direction to help participants know what they
should write in the :guilabel:`Multiple Lines Text Box` presented to them.

Constraints section
*******************

The :guilabel:`Constraints` section in the :guilabel:`Options` tab is the same, regardless of the
selected :guilabel:`Question Type`.

.. image:: time_random/constraints-section.png
   :align: center
   :alt: The Constraints section in the options tab in Odoo Surveys.

> In the :guilabel:`Constraints` section there is one option available: :guilabel:`Mandatory Answer`.

Reviewer: ```suggestion
In the :guilabel:`Constraints` section, there is one option available: :guilabel:`Mandatory Answer`.
```

When :guilabel:`Mandatory Answer` is enabled, that means that specific questions requires an answer
from the participant before they can move on. Also, when :guilabel:`Mandatory Answer` is enabled,
that reveals an additional field: :guilabel:`Error Message`.

> In the :guilabel:`Error Message` field, enter a custom message that appears to inform the
> participant that an answer is required for this specific question.

Reviewer: ```suggestion
In the :guilabel:`Error Message` field, enter a custom error message prompting the participant to provide an answer for this question.
```
